Tests/SonarScanner.MSBuild.Tasks.UnitTest/GetAnalyzerSettingsTests.cs

```diff
@@ -80,293 +63,217 @@ public void ConfigExistsButNoAnalyzerSettings_NoError()
     [DataRow("7.4")]
     public void ConfigExists_NoLanguage_SettingsOverwritten(string sonarQubeVersion)
     {
-        // Arrange
         var config = new AnalysisConfig
         {
             SonarQubeVersion = sonarQubeVersion,
-            AnalyzersSettings = new List<AnalyzerSettings>
-            {
+            AnalyzersSettings =
+            [
                 new AnalyzerSettings
                 {
                     Language = "cs",
                     RulesetPath = "f:\\yyy.ruleset",
-                    AnalyzerPlugins = new List<AnalyzerPlugin> { CreateAnalyzerPlugin("c:\\local_analyzer.dll") },
-                    AdditionalFilePaths = new List<string> { "c:\\add1.txt", "d:\\add2.txt", "e:\\subdir\\add3.txt" }
+                    AnalyzerPlugins = [CreateAnalyzerPlugin(Path.Combine("c:", "local_analyzer.dll"))],
+                    AdditionalFilePaths = [Path.Combine("c:", "add1.txt"), Path.Combine("d:", "add2.txt"), Path.Combine("e:", "subdir", "add3.txt")]
                 }
-            }
+            ]
         };
-
-        var testSubject = CreateConfiguredTestSubject(config, string.Empty /* no language specified */, TestContext);
-        testSubject.OriginalAdditionalFiles = new[]
-        {
+        var testSubject = CreateConfiguredTestSubject(config, string.Empty, TestContext);
+        testSubject.OriginalAdditionalFiles =
+        [
             "original.should.be.preserved.txt"
-        };
+        ];
 
-        // Act
         ExecuteAndCheckSuccess(testSubject);
 
-        // Assert
         testSubject.RuleSetFilePath.Should().BeNull();
         testSubject.AnalyzerFilePaths.Should().BeNull();
         testSubject.AdditionalFilePaths.Should().BeEquivalentTo("original.should.be.preserved.txt");
     }
 
-    [TestCategory(TestCategories.NoUnixNeedsReview)]
+    // SONARMSBRU-216: non-assembly files should be filtered out
     [TestMethod]
     public void ConfigExists_Legacy_SettingsOverwritten()
     {
-        // Arrange
-        // SONARMSBRU-216: non-assembly files should be filtered out
         var filesInConfig = new List<AnalyzerPlugin>
         {
-            CreateAnalyzerPlugin("c:\\analyzer1.DLL"),
+            CreateAnalyzerPlugin(Path.Combine("c", "analyzer1.dll")),
```

Comment:
If the tests fail on Linux with `c:`, make them Windows only tests.
```suggestion
            CreateAnalyzerPlugin(Path.Combine("c:", "analyzer1.dll")),
```

Reply:
What do you think about his:
https://github.com/SonarSource/sonar-scanner-msbuild/pull/2609#discussion_r2219511841